vendor/bat-native-ledger/src/bat/ledger/internal/publisher/server_publisher_fetcher.cc

```diff
@@ -0,0 +1,307 @@
+#include "base/strings/stringprintf.h"
+#include "base/time/time.h"
+#include "bat/ledger/internal/ledger_impl.h"
+#include "bat/ledger/internal/publisher/brotli_stream_decoder.h"
+#include "bat/ledger/internal/publisher/channel_response.pb.h"
+#include "bat/ledger/internal/publisher/prefix_util.h"
+#include "bat/ledger/internal/request/request_publisher.h"
+#include "bat/ledger/option_keys.h"
+#include "net/http/http_status_code.h"
+
+using std::placeholders::_1;
+using std::placeholders::_2;
+using std::placeholders::_3;
+
+namespace {
+
+constexpr size_t kQueryPrefixBytes = 2;
+
+int64_t GetCacheExpiryInSeconds(bat_ledger::LedgerImpl* ledger) {
+  DCHECK(ledger);
+  // NOTE: We are reusing publisher prefix list refresh interval for
+  // determining the cache lifetime of publisher details. At a later
+  // time we may want to introduce an additional option for this value.
+  return ledger->GetUint64Option(ledger::kOptionPublisherListRefreshInterval);
+}
+
+ledger::PublisherStatus PublisherStatusFromMessage(
+    const publishers_pb::ChannelResponse& response) {
+  switch (response.wallet_connected_state()) {
+    case publishers_pb::UPHOLD_ACCOUNT_KYC:
+      return ledger::PublisherStatus::VERIFIED;
+    case publishers_pb::UPHOLD_ACCOUNT_NO_KYC:
+      return ledger::PublisherStatus::CONNECTED;
+    default:
+      return ledger::PublisherStatus::NOT_VERIFIED;
+  }
+}
+
+ledger::PublisherBannerPtr PublisherBannerFromMessage(
+    const publishers_pb::SiteBannerDetails& banner_details) {
+  auto banner = ledger::PublisherBanner::New();
+
+  banner->title = banner_details.title();
+  banner->description = banner_details.description();
+
+  if (!banner_details.background_url().empty()) {
+    banner->background =
+        "chrome://rewards-image/" + banner_details.background_url();
+  }
+
+  if (!banner_details.logo_url().empty()) {
+    banner->logo = "chrome://rewards-image/" + banner_details.logo_url();
+  }
+
+  for (auto& amount : banner_details.donation_amounts()) {
+    banner->amounts.push_back(amount);
+  }
+
+  if (banner_details.has_social_links()) {
+    auto& links = banner_details.social_links();
+    if (!links.youtube().empty()) {
+      banner->links.insert(std::make_pair("youtube", links.youtube()));
+    }
+    if (!links.twitter().empty()) {
+      banner->links.insert(std::make_pair("twitter", links.twitter()));
+    }
+    if (!links.twitch().empty()) {
+      banner->links.insert(std::make_pair("twitch", links.twitch()));
+    }
+  }
+
+  return banner;
+}
+
+ledger::ServerPublisherInfoPtr ServerPublisherInfoFromMessage(
+    const publishers_pb::ChannelResponseList& message,
+    const std::string& expected_key) {
+  for (auto& entry : message.channel_responses()) {
+    if (entry.channel_identifier() != expected_key) {
+      continue;
+    }
+
+    // TODO(zenparsing): The previous JSON data had an "excluded"
+    // field, whereas the protobuf format does not. Do we still
+    // need this field?
```

Comment:
@NejcZdovc See the comment above. The `ChannelResponse` message doesn't contain the equivalent of an "excluded" field. Do we still need this field?

Reply:
It looks like "excluded" is used in `SaveVisitInternal` when recording activity for AC. Do we not need that logic anymore? Or do I need to request that the "excluded" field added to the new protocol?

Reply:
afaik we will have different list for excluded list @yachtcaptain23 to confirm

Reply:
We don't use the excluded list anymore. cc: @evq

Reply:
@NejcZdovc should I remove the "excluded" field from the table and associated mojom struct?

Reply:
@yachtcaptain23 how come that we don't have excluded list anymore? is everybody aware of this change? cc @mandar-brave

Reply:
@zenparsing yes remove excluded field and all related logic for this flow. Note that we still need excluded from user flow

Reply:
Removed `excluded` from `server_publisher_info` table and mojom structs.